src/cbdata.cc

```diff
@@ -373,7 +357,7 @@ cbdataInternalUnlock(const void *p)
 #if WITH_VALGRIND
     c = cbdata_htable.at(p);
 #else
-    c = (cbdata *) (((char *) p) - cbdata::Offset);
+    c = (cbdata *) (((char *) p) - offsetof(cbdata, data));
 #endif
```

Comment:
Please avoid duplicating this low-level code so many times. I would try to add cbdata::FromUserData(), FromCallerData(), or a similar method, replacing all 5 lines with:

```C++
c = cbdata::FromUserData(p);
```

Reply:
Done in 5b44429.